### PR TITLE
UI new list page

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

<what this pull request is doing>

- switch to react ui component library in new list page
- Use React components for add another meal modal of new list page
- use react components for recipe cost view of the add another meal modal

<what will be done in later PRs and not included here>

- implement react component library across entire app

## Milestones
<the milestones or stories/features that this works towards>

- implement react component library across entire app

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
https://docs.google.com/document/d/1u4QuX_0hLM_7-EKXGbdEtpxEErOixgLAIE5qFVQna7s/edit?usp=sharing

### CODE REVIEW

File: frontend/src/components/AddAnotherMealModal.tsx [hunk 1]
```diff
@@ -187,38 +161,60 @@ const AddAnotherMealModal: React.FC<GPAddAnotherMealProps> = ({
...
+                sx={{ my: 3, display: "flex", justifyContent: "space-between" }}
               >
-                Find some Recipes!
-              </Button>
+                {/* Code Referenced from MUI Documentation: https://mui.com/joy-ui/react-switch/ */}
+                <FormControl
+                  orientation="horizontal"
+                  sx={{ width: "20%", justifyContent: "space-between" }}
+                >
+                  <FormLabel>Use Dietary Preferences</FormLabel>
```
Comment (nithyad) [inline frontend/src/components/AddAnotherMealModal.tsx:172]: nit but maybe Apply Dietary Preferences is better than Use

File: frontend/src/components/AddAnotherMealModal.tsx [hunk 2]
```diff
@@ -76,8 +76,10 @@ const AddAnotherMealModal: React.FC<GPAddAnotherMealProps> = ({
...
-      : `https://api.spoonacular.com/recipes/complexSearch?apiKey=${API_KEY}&query=${recipeRequest}&number=${numToRequest}&addRecipeInformation=true&fillIngredients=true&offset=${offset}&instructionsRequired=true`;
+      ? `https://api.spoonacular.com/recipes/complexSearch?apiKey=${API_KEY}&query=${recipeRequest}&number=1&addRecipeInformation=true&fillIngredients=true&offset=${offset}&instructionsRequired=true&diet=${userDiets}&intolerances=${userIntolerances}`
+      : `https://api.spoonacular.com/recipes/complexSearch?apiKey=${API_KEY}&query=${recipeRequest}&number=1&addRecipeInformation=true&fillIngredients=true&offset=${offset}&instructionsRequired=true`;
+      // ? `https://api.spoonacular.com/recipes/complexSearch?apiKey=${API_KEY}&query=${recipeRequest}&number=${numToRequest}&addRecipeInformation=true&fillIngredients=true&offset=${offset}&instructionsRequired=true&diet=${userDiets}&intolerances=${userIntolerances}`
```
Comment (nithyad) [inline frontend/src/components/AddAnotherMealModal.tsx]: delete

File: frontend/src/components/AddAnotherMealModal.tsx [hunk 3]
```diff
@@ -76,8 +76,10 @@ const AddAnotherMealModal: React.FC<GPAddAnotherMealProps> = ({
...
+      // : `https://api.spoonacular.com/recipes/complexSearch?apiKey=${API_KEY}&query=${recipeRequest}&number=${numToRequest}&addRecipeInformation=true&fillIngredients=true&offset=${offset}&instructionsRequired=true`;
```
Comment (nithyad) [inline frontend/src/components/AddAnotherMealModal.tsx]: delete
